### DISCUSSION TITLE
add custom file parser function

### DISCUSSION THREAD

**Discussion by 7801943** [langgenius/dify#11325] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

with complex docx files，文档解析器 can't do parser well，so preceeding files by custom tools，apis，or code（in sandbox）need to be done

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by svcvit** [comment]
I have a relatively clever method; you can give it a try if needed. https://blog.vcvit.me/2024/12/05/use-dify-sandbox-to-parse-files/
